app/view_models/verify/base.rb

```diff
@@ -0,0 +1,88 @@
+module Verify
+  class Base
+    include Rails.application.routes.url_helpers
+
+    def initialize(error: nil, remaining_attempts:)
+      @error = error
+      @remaining_attempts = remaining_attempts
+    end
+
+    attr_reader :error, :remaining_attempts
+
+    def title
+      I18n.t("idv.titles.#{step_name}")
+    end
+
+    def modal_partial
+      if error.present?
+        'shared/modal_verification'
+      else
+        'shared/null'
+      end
+    end
+
+    def warning_partial
+      if error == 'warning'
+        'shared/modal_verification_warning'
+      else
+        'shared/null'
+      end
+    end
+
+    def message
+      html_paragraph(text: I18n.t("idv.modal.financials.#{error}")) if error
+    end
+
+    def button
```

Comment:
ultimately, I feel like this view-y stuff should be in a presenter, but perhaps as a follow up to this? I feel icky putting view-related code in a view model, but this refactor was already getting a tad larger than I wanted.